lib/grplatt.gi

```diff
@@ -2232,11 +2232,20 @@ InstallMethod(NormalSubgroups,"homomorphism principle perm groups",true,
 ##
 #M  Socle(<G>)
 ##
-InstallMethod(Socle,"from normal subgroups",true,[IsGroup],0,
+InstallMethod(Socle,"from normal subgroups",true,[IsGroup and IsFinite],0,
 function(G)
 local n,i,s;
   if Size(G)=1 then return G;fi;
```

Comment:
I know this is not part of your patch, but I wonder whether we should replace `Size(G)=1` by `IsTrivial(G)` here. It should never be worse, but in theory, it can handle some cases efficiently which `Size(G)=1` can not. Then again, those cases are mostly cases where G is huge, and for those cases, we plan on doing more complicated stuff below anyway...

So, probably don't do anyhing, and ignore me thinking out loud... :)